apps/backend/src/router/auth.ts

```diff
@@ -2,16 +2,55 @@ import { Request, Response, Router } from 'express';
 import passport from 'passport';
 import jwt from 'jsonwebtoken';
 import { db } from '../db';
+import { v4 as uuidv4 } from 'uuid';
+import { COOKIE_MAX_AGE } from '..';
 const router = Router();
 
 const CLIENT_URL =
   process.env.AUTH_REDIRECT_URL ?? 'http://localhost:5173/game/random';
 const JWT_SECRET = process.env.JWT_SECRET || 'your_secret_key';
 
+interface userJwtClaims {
+  userId: string;
+  name: string;
+  isGuest?: boolean;
+}
+
 interface User {
   id: string;
+  token?: string;
+  name: string;
+  isGuest?: boolean;
 }
 
+// this route is to be hit when the user wants to login as a guest
+router.post('/guest', async (req: Request, res: Response) => {
+  const bodyData = req.body;
+  let guestUUID = 'guest-' + uuidv4();
+
+  const user = await db.user.create({
+    data: {
+      username: guestUUID,
+      email: guestUUID + '@chess100x.com',
+      name: bodyData.name || guestUUID,
+      provider: 'GUEST',
+    },
+  });
+
+  const token = jwt.sign(
+    { userId: user.id, name: user.name, isGuest: true },
+    JWT_SECRET,
+  );
+  let User: User = {
```

Comment:
Whats going on here? can we change this?
use const and change User to something else

Reply:
we are returning userDetails so we can store it in the frontend state, changed the struct and variable names from "User" to "UserDetails"